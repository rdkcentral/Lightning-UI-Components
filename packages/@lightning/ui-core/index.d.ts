export * from './src/components';
export * from './src/mixins';
export * from './src/types/lui';
export * from './src/globals';

// exports without type definitions
export { default as utils } from './src/utils/index.js';
// export * from './src/shaders/index.js';
// export * from './src/textures/index.js';
