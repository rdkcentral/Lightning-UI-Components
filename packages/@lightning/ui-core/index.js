export { default as utils } from './src/utils/index.js';
export * from './src/components/index.js';
export * from './src/globals/index.js';
export * from './src/mixins/index.js';
export * from './src/shaders/index.js';
export * from './src/textures/index.js';
