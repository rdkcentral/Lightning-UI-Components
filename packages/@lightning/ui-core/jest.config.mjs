// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html
export default {
  // Automatically clear mock calls and instances between every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  collectCoverage: true,

  // An array of glob patterns indicating a set of files for which coverage information should be collected
  collectCoverageFrom: ['<rootDir>/src/*.js'],

  // The directory where Jest should output its coverage files
  coverageDirectory: 'coverage',

  // A list of reporter names that Jest uses when writing coverage reports
  coverageReporters: ['text', 'html'],

  // The maximum amount of workers used to run your tests. Can be specified as % or a number. E.g. maxWorkers: 10% will use 10% of your CPU amount + 1 as the maximum worker number. maxWorkers: 2 will use a maximum of 2 workers.
  maxWorkers: '4',

  moduleFileExtensions: ['js', 'json', 'jsx', 'node', 'ts', 'tsx', 'mjs'],

  // A map from regular expressions to module names that allow to stub out resources with a single module
  moduleNameMapper: {
    '\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/fileMock.js'
  },

  // TODO: Need to fix these tests after monorepo refactor
  modulePathIgnorePatterns: [
    '<rootDir>/src/mixins/withExtensions',
    '<rootDir>/src/components/Tooltip',
    '<rootDir>/src/mixins/withHandleKey',
    '<rootDir>/src/mixins/withLayout',
    '<rootDir>/src/globals/context'
  ],

  // The root directory that Jest should scan for tests and modules within
  rootDir: '.',

  // A list of paths to directories that Jest should use to search for files in
  roots: ['<rootDir>/src'],

  // The paths to modules that run some code to configure or set up the testing environment before each test
  setupFiles: ['jest-webgl-canvas-mock'],

  // The test environment that will be used for testing
  testEnvironment: 'jsdom',

  // Options that will be passed to the testEnvironment
  testEnvironmentOptions: { url: 'http://localhost', resources: 'usable' },

  // An array of regexp pattern strings that are matched against all test paths, matched tests are skipped
  testPathIgnorePatterns: [],

  // Ensure all is run as module
  transform: {},

  // Indicates whether each individual test should be reported during the run
  verbose: true // null,
};
