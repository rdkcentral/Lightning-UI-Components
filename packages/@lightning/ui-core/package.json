{
  "name": "@lightning/ui-core",
  "version": "1.5.1",
  "peerDependencies": {
    "@lightningjs/core": "^2.7.1"
  },
  "sideEffects": false,
  "type": "module",
  "exports": {
    ".": {
      "types": "./index.d.ts",
      "default": "./index.js"
    },
    "./storybook": {
      "default": "./storybook/index.js"
    }
  },
  "scripts": {
    "start": "yarn run generateStorybookData && yarn start-storybook --no-manager-cache -p 8000",
    "build": "rm -rf ./dist && rollup --config ../../../rollup/rollup.config.js --name lightning-ui-core",
    "lint": "eslint './src/**/*.js'",
    "lint:fix": "eslint --fix './src/**/*.js'",
    "test": "NODE_OPTIONS=--experimental-vm-modules jest --coverage --colors",
    "test:updateSnapshot": "NODE_OPTIONS=--experimental-vm-modules jest --updateSnapshot --colors",
    "test:watch": "NODE_OPTIONS=--experimental-vm-modules jest --no-coverage --watch --colors"
  },
  "devDependencies": {
    "@babel/core": "^7.19.3",
    "@lightning/ui-test-utils": "workspace:^",
    "@mdx-js/react": "^1.6.22",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "enhanced-resolve": "^5.10.0",
    "jest": "^29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "jest-webgl-canvas-mock": "^0.2.3",
    "lightning-ui-docs": "workspace:^",
    "rollup": "^3.2.5"
  },
  "dependencies": {
    "@suite-themes/base-lightning-tv": "^1.5.0",
    "debounce": "^1.2.1",
    "husky": "^8.0.2",
    "lint-staged": "^13.0.3"
  },
  "files": [
    "dist/*"
  ]
}
