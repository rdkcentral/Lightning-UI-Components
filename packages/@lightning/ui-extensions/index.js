export * from './src/FocusRing';
export * from './src/DropShadow';
