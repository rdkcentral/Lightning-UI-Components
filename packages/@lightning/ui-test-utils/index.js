export * from './src/lightning-test-renderer';
export * from './src/lightning-test-utils';
export * from './src/lng-test-env';
