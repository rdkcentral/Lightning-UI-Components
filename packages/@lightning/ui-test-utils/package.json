{
  "name": "@lightning/ui-test-utils",
  "version": "1.1.0",
  "type": "module",
  "exports": "./index.js",
  "files": [
    "src/*"
  ],
  "peerDependencies": {
    "@lightningjs/core": "^2.7.1"
  },
  "devDependencies": {
    "jest": "^29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "jest-webgl-canvas-mock": "^0.2.3"
  },
  "dependencies": {
    "@lightning/ui-core": "workspace:^",
    "@lightningjs/core": "^2.7.1"
  }
}
