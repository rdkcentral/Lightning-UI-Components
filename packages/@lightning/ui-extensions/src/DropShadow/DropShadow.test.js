import { makeCreateComponent } from '@lightning/ui-test-utils';
import dropShadowExtension, {
  dropShadowExtensionGenerator
} from './DropShadow.extension';
import { Base, withExtensions } from '@lightning/ui-core';

const width = 100;
const height = 150;

const generatedExtension = dropShadowExtensionGenerator({
  componentsToMask: ['ToMaskExample']
});

class Example extends Base {
  static get __componentName() {
    return 'Example';
  }
  static _template() {
    return {
      w: width,
      h: height,
      rect: true
    };
  }
}

class ToMaskExample extends Base {
  static get __componentName() {
    return 'ToMaskExample';
  }
  static _template() {
    return {
      w: width,
      h: height,
      rect: true
    };
  }
}

const createComponent = makeCreateComponent(
  dropShadowExtension(withExtensions(Example))
);

const createComponentGeneratedNoMask = makeCreateComponent(
  generatedExtension(withExtensions(Example))
);

const createComponentGeneratedWithMask = makeCreateComponent(
  generatedExtension(withExtensions(ToMaskExample))
);

describe('DropShadow Extension', () => {
  let component;
  let testRenderer;

  beforeEach(() => {
    [component, testRenderer] = createComponent();
  });

  afterEach(() => {
    component = null;
    testRenderer = null;
  });

  it('renders', () => {
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with generator', () => {
    [component, testRenderer] = createComponentGeneratedNoMask();
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds adds a DropShadow element to the component', () => {
    expect(component._DropShadow).toBeDefined();
  });

  it('Does not affect the size of the base component', () => {
    expect(component.w).toEqual(width);
    expect(component.h).toEqual(height);
  });

  it('Applies Masked shadow to the specified component', () => {
    [component, testRenderer] = createComponentGeneratedWithMask();
    expect(component._DropShadow.maskShadow).toBe(true);
  });

  it('Does not apply a mask to the unspecified component', () => {
    [component, testRenderer] = createComponentGeneratedNoMask();
    expect(component._DropShadow.maskShadow).toBeFalsy();
  });
});
