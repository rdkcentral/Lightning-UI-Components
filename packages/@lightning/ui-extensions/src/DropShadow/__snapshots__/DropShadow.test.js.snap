// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropShadow Extension renders 1`] = `
{
  "Component": {
    "DropShadow": {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": {
        "Frame": {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": {
            "Shadow": {
              "active": false,
              "alpha": 0,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": "99000000",
              "enabled": false,
              "flex": false,
              "flexItem": false,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Shadow",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": [
                "Shadow",
              ],
              "texture": {
                "type": "StaticCanvasTexture",
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 262,
          "isComponent": undefined,
          "mount": 0.5,
          "mountX": 0.5,
          "mountY": 0.5,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Frame",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "Frame",
          ],
          "visible": true,
          "w": 212,
          "x": 50,
          "y": 91,
          "zIndex": 0,
        },
      },
      "clipping": false,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 150,
      "hasFinalFocus": false,
      "hasFocus": false,
      "isComponent": true,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "DropShadow",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": [
        "DropShadow",
      ],
      "type": "Shadow",
      "visible": true,
      "w": 100,
      "x": 0,
      "y": 0,
      "zIndex": -1,
    },
  },
}
`;

exports[`DropShadow Extension renders with generator 1`] = `
{
  "Component": {
    "DropShadow": {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": {
        "Frame": {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": {
            "Shadow": {
              "active": false,
              "alpha": 0,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": "99000000",
              "enabled": false,
              "flex": false,
              "flexItem": false,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Shadow",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": [
                "Shadow",
              ],
              "texture": {
                "type": "StaticCanvasTexture",
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 262,
          "isComponent": undefined,
          "mount": 0.5,
          "mountX": 0.5,
          "mountY": 0.5,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Frame",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "Frame",
          ],
          "visible": true,
          "w": 212,
          "x": 50,
          "y": 91,
          "zIndex": 0,
        },
      },
      "clipping": false,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 150,
      "hasFinalFocus": false,
      "hasFocus": false,
      "isComponent": true,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "DropShadow",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": [
        "DropShadow",
      ],
      "type": "Shadow",
      "visible": true,
      "w": 100,
      "x": 0,
      "y": 0,
      "zIndex": -1,
    },
  },
}
`;
